Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Base stage: Install dependencies
+FROM python:3.10-slim AS base
+
+# Install system dependencies
+RUN apt-get update && \
+    apt-get install -y \
+    build-essential \
+    gcc \
+    libssl-dev \
+    poppler-utils \
+    tesseract-ocr \
+    tesseract-ocr-eng \
+    tesseract-ocr-kor && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set work directory
+WORKDIR /usr/src/app
+
+# Copy project files
+COPY . /usr/src/app
+
+# Install Python dependencies
+RUN pip install --upgrade pip setuptools setuptools-scm
+RUN pip install -r requirements.txt
+
+# Test stage: Run tests if CI=true
+FROM base AS test
+
+# Install testing dependencies
+RUN pip install pytest pytest-xdist
+
+# Run tests if CI is set to true
+RUN pytest -o log_cli=true --log-cli-level=INFO -n auto tests
+
+# Production stage: Create final image for production
+FROM base AS production
+
```

Comment:
Can you add option for the huggingface .cache folder?
In AutoRAG, we uses many models like T5, UPR, TART and so on.
And most of the model files stored in the `~/.cache/huggingface`

If there is already downloaded models in the local (outside container), it will be better to mount it to the docker container cache. 

Plus, we can adjust huggingface cache folder using env variable `HF_HOME` [refer here](https://huggingface.co/docs/huggingface_hub/ko/package_reference/environment_variables)

So two options here I think;
1. mount outside huggingface cache folder to directly docker container ~/.cache/huggingface
2. mount elsewhere in the docker container and set `HF_HOME` variable.

Reply:
It will be great you add it at README.md also.